EIPS/eip-3455.md

```diff
@@ -0,0 +1,73 @@
+---
+eip: 3455
+title: SUDO Opcode
+author: William Morriss (@wjmelements), Baptiste Vauthey (@thabaptiser)
+discussions-to: https://ethereum-magicians.org/t/eip-3455-sudo-opcode/5860
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-04-01
+---
+
+## Simple Summary
+A new opcode is introduced to allow calling from an arbitrary sender address.
+
+## Abstract
+A new opcode, `SUDO`, is introduced with the same parameters as `CALL`, plus another parameter to specify the sender address.
+
+## Motivation
+There are many use cases for being able to set the sender.
+
+Many tokens are stuck irretrievably because nobody has the key for the owner address.
+In particular, at address zero there is approximately 17 billion USD in tokens and ether according to [etherscan](https://etherscan.io/address/0x0000000000000000000000000000000000000000).
+With `SUDO`, anyone could free that value, leading to an economic boom that would end poverty and world hunger.
+Instead it is sitting there idle like the gold in Fort Knox.
+`SUDO` fixes this.
+
+It is a common mistake to send ERC20 tokens to the token address instead of the intended recipient.
+This happens because users paste the token address into the recipient fields.
+Currently there is no way to recover these tokens.
+`SUDO` fixes this.
+
+Many scammers have fraudulently received tokens and ETH via trust-trading.
+Their victims currently have no way to recover their funds.
+`SUDO` fixes this.
+
+Large amounts of users have accidentally locked up tokens and ether by losing their private keys.
+This is inefficient and provides a bad user experience.
+To accommodate new and inexperienced users, there needs to be a way to recover funds after the private key has been lost.
+`SUDO` fixes this.
+
+Finally, there are many tokens and ether sitting in smart contracts locked due to a bug.
+We could finally close [#156](https://github.com/ethereum/EIPs/issues/156).
+We cannot currently reclaim ether from stuck accounts.
+`SUDO` fixes this.
+
+## Specification
+Adds a new opcode (`SUDO`) at `0xf8`.
+`SUDO` pops 8 parameters from the stack.
```

Comment:
I like this proposal but I'm worried about the security implications of this. If it gets included as it is, everyone would be able to move funds around!

It would be much more secure if the `SUDO` opcode pops 9 parameters from the stack, where the first one is `password`. My bank doesn't allow passwords longer than 8 bytes and as we know banks are really secure, so it's safe to assume we need only 64 bits for the `password` field.

Reply:
Don't forget about the PIN! We could use the upper 96 bits of the Sender address field for the password, with 64 bits for the password and the other 32 for a 4-digit PIN. This would address any security concerns.

Reply:
It fits perfectly actually, we can squeeze everything in 256 bits. To recap:

- `64 bits`: 8 characters password
- `32 bits`: 4-digit pin
- `160 bits`: address

Reply:
What about 2FA? Not sure if it's really safe without.

Reply:
My bank doesn't use 2fa so I don't think that would help much. Besides, the PIN and the password are already two factors.

Reply:
I think it is good security hygene to use 2FA though. How about we set a field called `2FA` and hardcode it to `passed`? That way everyone is happy.

Reply:
We'll be using FIFO, right?

Reply:
@vrde this is unnecessary complication. Ethereum does not have security as a primary value, let others worry about it. It's much more important for ETH to focus on being world computer and a computer must first and foremost work. Adding security features increases code complexity and unreliability.

`sudo mine_this_transaction` would be much more useful than worrying about security since filthy capitalist miners are ripping off poor ETH users on fees and cause unreliability. Working class can't use the computer today, only the rich can.